Assets/_BForBoss/_Leaderboard/Scripts/Username/InputUsernameViewModel.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class InputUsernameViewModel
+    {
+        private readonly InputUsername _inputUsername = new InputUsername();
+
+        public event Action OnSuccess;
+        public event Action OnFailure;
+
+        public string Username => _inputUsername.Username;
+
+        public void SetUserName(string username)
+        {
+            if (_inputUsername.CanUseThisUsername(username))
+            {
+                _inputUsername.SetUserName(username);
+                OnSuccess?.Invoke();
+            }
+            else
+            {
+                OnFailure?.Invoke();
+            }
+        }
+
+        public bool IsUsernameAlreadySet()
+        {
+            return PlayerPrefs.HasKey(UploadPlayerScoreDataSource.PlayerPrefKey.UserName);
+        }
+    }
```

Comment:
```suggestion
    public void RemoveSubscribers()
    {
        OnSuccess = null;
        OnFailure = null;
    }
    
```

Reply:
I would think it's better to have the subscribers unsubscribe manually. This gives it a false idea that we wouldn't need to unsubscribe